From 21b746743c146c94235c93611519db6d47c8c0c1 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Thu, 18 Feb 2021 11:01:31 +0100
Subject: [PATCH] fix(fcoe): rd.nofcoe=0 should disable fcoe

8446c8f9 Changed the default behavior, but also flipped meaning of 0/1.
Right now rd.nofcoe=0 enables fcoe, which is the opposite what manpage
says.

(cherry picked from commit 805b46c2a81e04d69fc3af912942568516d05ee7)

Resolves: #1929201
---
 modules.d/95fcoe/lldpad.sh     | 2 +-
 modules.d/95fcoe/parse-fcoe.sh | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/95fcoe/lldpad.sh b/modules.d/95fcoe/lldpad.sh
index 7faa2e19..6b1844b0 100755
--- a/modules.d/95fcoe/lldpad.sh
+++ b/modules.d/95fcoe/lldpad.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-if getargbool 0 rd.nofcoe ; then
+if ! getargbool 1 rd.nofcoe ; then
 	info "rd.nofcoe=0: skipping lldpad activation"
 	return 0
 fi
diff --git a/modules.d/95fcoe/parse-fcoe.sh b/modules.d/95fcoe/parse-fcoe.sh
index 86f4331e..5553602f 100755
--- a/modules.d/95fcoe/parse-fcoe.sh
+++ b/modules.d/95fcoe/parse-fcoe.sh
@@ -13,7 +13,7 @@
 # fcoe=eth0:nodcb:vn2vn
 # fcoe=4a:3f:4c:04:f8:d7:nodcb:fabric
 
-if getargbool 0 rd.nofcoe ; then
+if ! getargbool 1 rd.nofcoe ; then
 	info "rd.nofcoe=0: skipping fcoe"
 	return 0
 fi

